core/src/com/unciv/logic/GameSaver.kt

```diff
@@ -29,7 +29,10 @@ class GameSaver {
     }
 
     fun saveGame(game: GameInfo, GameName: String, multiplayer: Boolean = false) {
-        json().toJson(game,getSave(GameName, multiplayer))
+        var name = GameName
+        if (multiplayer)
+            name = addGame(game.gameId, name)
```

Comment:
addGame and removeGame are not indicative - that is to say, it's not clear that we're looking at the cache here, since it's in a companion object.
Moving it to an object class of its own would make this MultiplayerFileManager.addGame(), which is better, and changing it to MultiplayerInfoCache (or something along those lines) would make it even clearer what this line is doing

Reply:
Also, I know this doesn't change the functionality, but this line should probably be after the actual saving (first save the game, then update the cache)